core/src/test/java/org/infinispan/configuration/UpdatableConfigurationTest.java

```diff
@@ -0,0 +1,30 @@
+package org.infinispan.configuration;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName= "configuration.UpdatableConfigurationTest")
+public class UpdatableConfigurationTest {
+
+   public void testUpdateConfiguration() {
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.expiration().maxIdle(1000).lifespan(2000);
+      Configuration configuration = builder.build();
+      builder = new ConfigurationBuilder();
+      builder.expiration().maxIdle(3000);
+
+      // Check that attributes are correct before the update
+      assertEquals(1000, configuration.expiration().maxIdle());
+      assertEquals(2000, configuration.expiration().lifespan());
+
+      configuration.update(builder.build());
+
+      // Check that only the modified attributes have been updated
+      assertEquals(3000, configuration.expiration().maxIdle());
+      assertEquals(2000, configuration.expiration().lifespan());
+   }
+
```

Comment:
The following fails due to the `isModified()` check in `Attribute.java`.

```java
   public void testUpdateToDefaultAttributeValue() {
      ConfigurationBuilder builder = new ConfigurationBuilder();
      builder.expiration().maxIdle(1000);
      Configuration configuration = builder.build();

      // Check that attributes are correct before the update
      assertEquals(1000, configuration.expiration().maxIdle());

      configuration.update(new ConfigurationBuilder().build());

      // Check that the attributes have been updated to use the default value
      assertEquals(-1, configuration.expiration().maxIdle());
   }
```

Reply:
Fixing the method as per-your suggestion applies the "default" behaviour to the lifespan attribute in the test I already had there. I've fixed that instead of including your test method here which would be redundant.